platform/ontology/api.go

```diff
@@ -45,99 +47,165 @@ func (p *Platform) GetTokenTxsByAddress(address string, token string) (blockatla
 		height = (int64)(block.Result[0].Height)
 	}
 	return height, nil
 }
 
-func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
-	response, err := p.client.GetBlockByNumber(num)
+func (p *Platform) GetBlockByNumber(num int64) (block *blockatlas.Block, err error) {
+	blockOnt, err := p.client.GetBlockByNumber(num)
+	if err != nil {
+		return nil, err
+	}
+	txsRaw, err := p.getTxDetails(blockOnt.Result.TxnList)
 	if err != nil {
-		logger.Error("GetBlockByNumber", logger.Params{"platform": p.Coin().Symbol, "details": err.Error()})
 		return nil, err
 	}
-	var (
-		block blockatlas.Block
-		txs   []blockatlas.Tx
-	)
-	if response.Error == 0 {
-		block.ID = response.Result.Hash
-		block.Number = int64(response.Result.Height)
-		for _, txn := range response.Result.TxnList {
-			tx := new(blockatlas.Tx)
-			tx.ID = txn.TxnHash
-			tx.Block = uint64(txn.Height)
-			if txn.ConfirmFlag == 1 {
-				tx.Status = blockatlas.StatusCompleted
+
+	block = normalizeBlock(*blockOnt, txsRaw)
+	return
+}
+
+func (p *Platform) getTxDetails(txnList []Tx) ([]TxV2, error) {
+	var wg sync.WaitGroup
+	txsChan := make(chan TxV2, len(txnList))
+	wg.Add(len(txnList))
+	for _, blockTxRaw := range txnList {
+		go func(blockTxRaw Tx, wg *sync.WaitGroup) {
+			defer wg.Done()
+			txRaw, err := p.client.GetTxDetailsByHash(blockTxRaw.TxnHash)
+			if err == nil {
+				txsChan <- txRaw
 			}
-			tx.Date = int64(txn.TxnTime)
-			tx.Coin = coin.Ontology().ID
-			txs = append(txs, *tx)
+		}(blockTxRaw, &wg)
+	}
+	wg.Wait()
+	close(txsChan)
+	if len(txsChan) != len(txnList) {
+		return nil, errors.E("getTxDetails failed to call client.GetTxDetailsByHash http get or unmarshal")
+	}
+	var txsRaw []TxV2
+	for tx := range txsChan {
+		if len(tx.TxDetails.Transfers) > 0 {
+			txsRaw = append(txsRaw, tx)
 		}
-		block.Txs = txs
 	}
-
-	return &block, nil
+	return txsRaw, nil
 }
 
-func Normalize(srcTx *Tx, assetName string) (tx blockatlas.Tx, ok bool) {
-	if len(srcTx.TransferList) < 1 {
-		return tx, false
+func normalizeBlock(blockOnt BlockResult, txsRaw []TxV2) (block *blockatlas.Block) {
+	block = &blockatlas.Block{
+		Number: int64(blockOnt.Result.Height),
+		ID:     blockOnt.Result.Hash,
 	}
-	transfer := srcTx.TransferList[0]
-	fee := numbers.DecimalExp(srcTx.Fee, 9)
-	var status blockatlas.Status
-	if srcTx.ConfirmFlag == 1 {
-		status = blockatlas.StatusCompleted
-	} else {
-		status = blockatlas.StatusFailed
+	block.Txs = normalizeBlockTransactions(txsRaw)
+	return
+}
+
+func normalizeBlockTransactions(txsRaw []TxV2) (txs []blockatlas.Tx) {
+	if len(txsRaw) > 0 {
+		for _, txRaw := range txsRaw {
+			tx := normalizeBlockTransaction(txRaw)
+			txs = append(txs, *tx)
+		}
 	}
+	return
+}
 
-	tx = blockatlas.Tx{
-		ID:     srcTx.TxnHash,
+func normalizeBlockTransaction(txRes TxV2) *blockatlas.Tx {
+	fee := numbers.DecimalExp(txRes.Fee, ONGDecimalsAmount)
+	tx := blockatlas.Tx{
+		ID:     txRes.TxHash,
 		Coin:   coin.ONT,
+		Date:   int64(txRes.TxTime),
```

Comment:
suggestion to make `txRes.TxTime` as int64 already? or make Date as `uint64 `

 same for block below?